### ISSUE TITLE
Segmentation Fault with pvr.demo

### ISSUE THREAD

**Issue by DeltaMikeCharlie** [kodi-pvr/pvr.demo#161]
Downloaded and built `Kodi`, `pvr.hts` and `pvr.demo` from source on Ubuntu using the following directions.

https://github.com/xbmc/xbmc/blob/master/docs/README.Linux.md
https://github.com/kodi-pvr/pvr.demo
https://github.com/kodi-pvr/pvr.hts

Debugging symbols were enabled for Kodi.

Kodi by itself executes without incident.
Kodi + pvr.hts executes without incident.
Kodi + pvr.demo results in a segmentation fault.

gdb output:

```
Thread 45 "PVRManager" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fff58ff9640 (LWP 144219)]
0x0000555557f944f3 in PVR::CPVRClient::UpdateTimerTypes()::{lambda(AddonInstance_PVR const*)#1}::operator()(AddonInstance_PVR const*) const ()

```
Could this be related to this recent Kodi PR: https://github.com/xbmc/xbmc/pull/25552 ?

Could `pvr.demo` be providing a more complete suite of properties than `pvr.hts` leading to this error?

**Comment by fuzzard** [comment]
Most likely fixed by [#25581](https://github.com/xbmc/xbmc/pull/25581)

**Comment by DeltaMikeCharlie** [comment]
> Most likely fixed by [#25581](https://github.com/xbmc/xbmc/pull/25581)

Agreed, although I had not seen that PR.  Thanks for bringing it to my attention.

I had added copious logging and had isolated the lines involved.  They match the lines mentioned in the PR.

**Comment by DeltaMikeCharlie** [comment]
I applied the changes from the proposed PR manually and the issue was resolved.  I will wait for the PR to be merged and go from here.
